hedera-node/docs/design/services/smart-contract-service/hbar-allowance-approval.md

```diff
@@ -0,0 +1,97 @@
+# hbar Allowance and Approval
+
+## Purpose
+
+The current IHederaTokenService interface within the Hedera Smart Contract Service provides support for approving an allowance to a `spender` account for tokens. This document outlines the implementation of hbar allowance and approval functionality in the Hedera Smart Contract Service, extending this capability to an account's hbars.
+
+## Goals
+
+The goal is to define the necessary classes for implementing the `hbarAllowance` and `hbarApprove` functionality within the Hedera Smart Contract Service. We'll focus on the implementation details in the modularized code base.
+## Non Goals
+
+The implementation for crypto allowance and approval in the Token Service Module will only be described at a high level, as this implementation is already completed. The smart contract service will dispatch the transaction to the Token Service Module for processing.
+
+## Architecture
+
+From an end-user perspective, we introduce two new interfaces to support the `hbarAllowance` and `hbarApprove` functionality:
+
+### IHRC632
```

Comment:
We were trying to follow Ethereum EIP/ERC conventions.  For example the ERC20 standard which has an `IERC20` interface was introduced in EIP 20.  Since HAS was introduced in HIP 632, we followed the same pattern to come up with `IHRC632`

Reply:
Either that or we explore interface inheritance which I've been thinking on and say `IHRC906` and `IHRC632`
are separate interfaces and come together like this

```solidity
interface IHRC906 {
    function hbarAllowance(address owner, address spender) external returns (uint256 responseCode);
    function hbarApprove(address owner, address spender, int256 amount) external returns (uint256 responseCode);
}

interface IHRC632 {
    function isAuthorized(address account, bytes memory messageHash, bytes memory signatureBlob) external returns (bool authorizedResponse);
    function isAuthorizedRaw(address account, bytes memory messageHash, bytes memory signatureBlob) external returns (bool authorizedResponse);
}

abstract contract IHederaAccountService is IHRC906, IHRC632 {
}
```

Reply:
Well, that was another thought I had - interface inheritance - but the reason I didn't suggest it was because the `IHederaAccountService` methods for `hbarAllowance` and `hbarApprove` have different arguments - and it isn't just an issue of overloading, which is fine, but the issue is who's doing the calling is different IIUC.  In the `IHederaAccountService` case it's some contract calling to get/set allowances on some other account, and in the `IHRC906` case it's us automagically calling the method on behalf of the EOA. Right?  (Or not?)

Reply:
`IHRC906` is only automagically when called via an EOA directly. When called within contacts it's straight forward like the other system contract function and requires you be explicit about all the entities involved